source/blog/2018-03-20-new-rpm-ostree-features.html.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Latest rpm-ostree features
+author: jlebon
+date: 2018-03-20 17:00:00 UTC
+tags: rpm-ostree, rpm, ostree, atomic, atomic host
+comments: true
+published: true
+---
+
+[rpm-ostree](https://rpm-ostree.readthedocs.org/) is the
+hybrid image/package system that provides transactional
+upgrades on Atomic Host.
+
+This blog post is a high-level summary of the recent
+features that were added in the last few releases.
+
+READMORE
+
+### New rojig delivery mechanism
+
+By far the most fundamental shift that has occurred in the
+latest releases is support for a different way of delivering
+OSTrees to client machines. Dubbed "rojig", the idea is that
+rather than clients downloading commits from OSTree remotes,
+they instead re-assemble the exact same commit using RPMs
+from yum repos. How? When OSTree commits are created on the
+server, most of its files can be traced back to specific
+RPMs. The work of figuring out which files come from which
+RPM is done on the compose server. Files which do not come
+from an RPM (e.g. the initramfs) are dumped in a special
+"rojig RPM". Using this rojig RPM and all other constituent
+RPMs, which are in essence fancy archives, the client is
+capable of reproducing the bit-for-bit matching OSTree
+commit.
+
+This allows content providers to only worry about delivering
+RPMs, rather than yet another format. It opens the doors to
+the many established tools and services that exist for
+organizing and distributing RPMs.
+
+You can try this out today using the instructions in
+[FAHC](https://pagure.io/fedora-atomic-host-continuous/):
+
+```
+$ rpm-ostree status
+State: idle; auto updates disabled
+Deployments:
+● ostree://fedora-atomic:fedora/27/x86_64/atomic-host
+                   Version: 27.100 (2018-03-13 17:19:44)
+                    Commit: 326f62b93a5cc836c97d31e73a71b6b6b6955c0f225f7651b52a693718e6aa91
+              GPGSignature: Valid signature by 860E19B0AFA800A1751881A6F55E7430F5282EE4
+$ cat >/etc/yum.repos.d/fahc.repo <<EOF
+[fahc]
+baseurl=https://ci.centos.org/artifacts/sig-atomic/fahc/jigdo
+gpgcheck=0
+EOF
+$ rpm-ostree rebase --experimental rojig://fahc:fedora-atomic-host
+...
+Will download: 463 packages (349.5 MB)
+  Downloading from updates: [=============] 100%
+  Downloading from fedora: [=============] 100%
+  Downloading from fahc: [=============] 100%
+Importing (463/463) [=============] 100%
+...
+$ rpm-ostree status -v
+State: idle; auto updates disabled
+Deployments:
+  fahc:fedora-atomic-host-27.406-1.fc27.x86_64
```

Comment:
@jlebon shouldn't this say `rojig://fahc:fedora-atomic-host-27.406-1.fc27.x86_64`. Did we change the behavior in the code?

Reply:
Right, that change was discussed in https://github.com/projectatomic/rpm-ostree/pull/1240.